src/main/java/seedu/address/model/person/Event.java

```diff
@@ -0,0 +1,112 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Objects;
+
+import seedu.address.commons.util.DateUtil;
+import seedu.address.commons.util.TimeUtil;
+
+/**
+ * Represents a Date event for a Person in the FriendDex.
+ */
+public class Event {
+    private final LocalDate date;
+    private final LocalTime time;
+    private final String description;
+    private final boolean hasTime;
+
+    /**
+     * Constructs a {@code Event}
+     *
+     * @param date A valid date.
+     * @param description A description of the event.
+     */
+    public Event(LocalDate date, String description) {
+        requireAllNonNull(date, description);
+        this.date = date;
+        this.time = null;
+        this.description = description;
+        this.hasTime = false;
+    }
+
+    /**
+     * Constructs a {@code Event}
+     *
+     * @param date A valid date.
+     * @param time A valid time.
+     * @param description A description of the event.
+     */
+    public Event(LocalDate date, LocalTime time, String description) {
+        requireAllNonNull(date, time, description);
+        this.date = date;
+        this.time = time;
+        this.description = description;
+        this.hasTime = true;
+    }
+
+    public LocalDate getDate() {
+        return date;
+    }
+
+    public LocalTime getTime() {
+        return time;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public boolean hasTime() {
+        return hasTime;
```

Comment:
`return time == null`, no need to store another instance variable.

Reply:
I was thinking it could be better as an abstraction, providing a easy way for other classes using the Event class to perform the check, as it doesn’t expose the inner workings of the Event class.